core/model/modx/modx.class.php

```diff
@@ -1935,15 +1935,24 @@ public function removeAllEventListener() {
      *
      * @param string $event Name of the event.
      * @param integer $pluginId Plugin identifier to add to the event.
+     * @param string $propertySetName The name of property set bound to the plugin
      * @return boolean true if the event is successfully added, otherwise false.
      */
-    public function addEventListener($event, $pluginId) {
+    public function addEventListener($event, $pluginId, $propertySetName = '') {
         $added = false;
+        $pluginId = intval($pluginId);
         if ($event && $pluginId) {
             if (!isset($this->eventMap[$event]) || empty ($this->eventMap[$event])) {
                 $this->eventMap[$event]= array();
             }
-            $this->eventMap[$event][]= $pluginId;
+            $ElementPropertySetTbl= $this->getTableName('modElementPropertySet');
+            $propsetTbl= $this->getTableName('modPropertySet');
+            $sql= "SELECT PropertySet.name 
+                FROM {$ElementPropertySetTbl} ElementPropSet 
+                LEFT JOIN {$propsetTbl} PropertySet ON ElementPropSet.property_set = PropertySet.id 
+                WHERE ElementPropSet.element = {$pluginId} AND ElementPropSet.element_class='modPlugin' AND PropertySet.name = '{$propertySetName}' ";
+            $propertySetName = $this->getValue($this->prepare($sql));
```

Comment:
I agree with @rtripault. A plugin should be registered by that method to an event listener on the fly, the property set should be selectable there on the fly too.